app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,27 +10,49 @@ def self.call(*args)
       end
 
       def call
-        return if commented_on_welcome_thread? || received_notification?
+        return unless user
+
+        send_welcome_notification
+        send_authentication_notification
+      end
+
+      def send_welcome_notification
+        return if received_notification?(welcome_broadcast) || commented_on_welcome_thread?
 
         Notification.send_welcome_notification(user.id, welcome_broadcast.id)
       end
 
-      def received_notification?
-        Notification.exists?(notifiable: welcome_broadcast, user: user)
+      def send_authentication_notification
+        return if received_notification?(authentication_broadcast) || authenticated_with_both_services? || user.created_at > 1.day.ago
+
+        Notification.send_welcome_notification(user.id, authentication_broadcast.id)
+      end
+
+      private
+
+      attr_reader :user
+
+      def received_notification?(broadcast)
+        Notification.exists?(notifiable: broadcast, user: user)
       end
 
       def commented_on_welcome_thread?
         welcome_thread = Article.admin_published_with("welcome").first
         Comment.where(commentable: welcome_thread, user: user).any?
       end
 
-      private
+      def authenticated_with_both_services?
+        user.identities.exists?(provider: "github") && user.identities.exists?(provider: "twitter")
+      end
 
       def welcome_broadcast
         @welcome_broadcast ||= Broadcast.find_by(title: "Welcome Notification: welcome_thread")
       end
 
-      attr_reader :user
+      def authentication_broadcast
+        missing_identity = user.identities.exists?(provider: "github") ? "twitter_connect" : "github_connect"
```

Comment:
I think this logic wouldn't be quite right if someone had authenticated with both services, right? 🤔

Reply:
If a user is authenticated with both services, `authenticated_with_both_services?` should have caught it. That said, I will think of a counter measure

Reply:
outdated.